src/HomeKit/HMEventTrigger.cs

```diff
@@ -24,7 +26,10 @@ partial class HMEventTrigger {
 #endif
 		static public NSPredicate CreatePredicateForEvaluatingTriggerOccurringBeforeSignificantEvent (HMSignificantEvent significantEvent, NSDateComponents offset)
 		{
-			return CreatePredicateForEvaluatingTriggerOccurringBeforeSignificantEvent (significantEvent.GetConstant (), offset);
+			var constant = significantEvent.GetConstant ();
+			if (constant is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (significantEvent));
+			return CreatePredicateForEvaluatingTriggerOccurringBeforeSignificantEvent (constant, offset);
```

Comment:
This method and `CreatePredicateForEvaluatingTriggerOccurringAfterSignificantEvent` below both eventually throw an ArgumentNullException so we could `!` the `significantEvent.GetConstant ()` and just pass it on but I thought this may be better. Opinions welcome!